addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/handler/SleepIQCloudHandler.java

```diff
@@ -0,0 +1,322 @@
+            int pollingInterval = getConfigAs(SleepIQCloudConfiguration.class).pollingInterval;
+            pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, pollingInterval, pollingInterval,
+                    TimeUnit.SECONDS);
+        } else if (bedStatusListeners.isEmpty() && pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    /**
+     * Retrieve the latest status on all beds and update all registered listeners.
+     */
+    private void publishBedStatusUpdates() {
+        publishBedStatusUpdates(null);
+    }
+
+    /**
+     * Retrieve the latest status on all beds and update the given listener. If the listener is <code>null</code>, all
+     * registered listeners will be updated.
+     *
+     * @param listener the listener to update (may be <code>null</code>)
+     */
+    private void publishBedStatusUpdates(final BedStatusListener listener) {
+
+        SleepIQ cloud = getCloud();
+        if (cloud == null) {
+            return;
+        }
+
+        FamilyStatus status = cloud.getFamilyStatus();
+        for (BedStatus bedStatus : status.getBeds()) {
+
+            if (listener != null) {
+                listener.onBedStateChanged(cloud, bedStatus);
+            } else {
+                bedStatusListeners.stream().forEach(l -> l.onBedStateChanged(cloud, bedStatus));
+            }
+        }
+    }
+
+    /**
+     * Register the given listener to receive bed status updates.
+     *
+     * @param listener the listener to register
+     */
+    public void registerBedStatusListener(final BedStatusListener listener) {
+
+        if (listener == null) {
+            return;
+        }
+
+        bedStatusListeners.add(listener);
+        publishBedStatusUpdates(listener);
+        updateListenerManagement();
+    }
+
+    /**
+     * Unregister the given listener from further bed status updates.
+     *
+     * @param listener the listener to unregister
+     * @return <code>true</code> if listener was previously registered and is now unregistered; <code>false</code>
+     *         otherwise
+     */
+    public boolean unregisterBedStatusListener(final BedStatusListener listener) {
+
+        boolean result = bedStatusListeners.remove(listener);
+        if (result) {
+            updateListenerManagement();
+        }
+
+        return result;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // all channels are read-only
+    }
+
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+
+        Collection<ConfigStatusMessage> configStatusMessages = new ArrayList<>();
+
+        SleepIQCloudConfiguration config = getConfigAs(SleepIQCloudConfiguration.class);
+        String username = config.username;
+        String password = config.password;
+
+        if (username == null || username.isEmpty()) {
```

Comment:
Minor: You could consider using commons StringUtils, their isEmpty check is null-safe